### PR TITLE
Migrate Makefile to GitHub actions

### PR DESCRIPTION
**Pull request by kdowney-cloudwallcapital**
SERENITY-1005

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+on: [push]
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    env:
+      SRC_DIR: src/python
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8"]
```
Comment (bmaycc) [inline .github/workflows/ci.yml:12]: very cool

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -31,7 +39,7 @@ jobs:
       run: poetry self add poetry-bumpversion
     - name: Build Package
       run: |
-        poetry version ${{ github.event.inputs.releaseVersion }}
+        poetry version $(git describe --tags --abbrev=0)
```
Comment (bmaycc) [inline .github/workflows/publish.yml:42]: curious, wouldn't **new_tag** or **new_version** be good inputs to this, instead of looking at the repo?

**Review by bmaycc — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (kdowney-cloudwallcapital) [reply .github/workflows/publish.yml:42]: This is how the version earlier today worked: it tagged the release automatically on every merge to master, and then it was a manual operation to publish and you had to name the tag. But I changed it up to do the tag and publish as a single, manually-triggered workflow. For this reason the tag is only known at runtime, and cannot be an input, but it then guarantees that the latest tagged version is what you publish.

Comment (bmaycc) [reply .github/workflows/publish.yml:42]: I totally get that, but isn't this the tag that was generated by the step in line 10 of the action, in which case the output of the step would be the poetry input here?

**Pull request merged**